Compile Error with GCC 8.3.0 (MSYS2/MinGW-w64)

NOTE: it's possible you're already aware of this and/or that the issue has been resolved.

Building the tests for PEGTL fails with the following error message after running CMake and calling `make`:

```
Scanning dependencies of target pegtl-test-utf8_general
[  0%] Building CXX object src/test/pegtl/CMakeFiles/pegtl-test-utf8_general.dir/utf8_general.cpp.o
In file included from D:/mingw-distro-setup/src/include/tao/pegtl/mmap_input.hpp:22,
                 from D:/mingw-distro-setup/src/include/tao/pegtl/file_input.hpp:16,
                 from D:/mingw-distro-setup/src/include/tao/pegtl.hpp:24,
                 from D:/mingw-distro-setup/src/src/test/pegtl/test.hpp:13,
                 from D:/mingw-distro-setup/src/src/test/pegtl/utf8_general.cpp:4:
D:/mingw-distro-setup/src/include/tao/pegtl/internal/file_mapper_win32.hpp: In constructor 'tao::pegtl::internal::file_mapper::file_mapper(const tao::pegtl::internal::win32_file_mapper&)':
D:/mingw-distro-setup/src/include/tao/pegtl/internal/file_mapper_win32.hpp:150:80: error: type qualifiers ignored on cast result type [-Werror=ignored-qualifiers]
                                                                            0 ) ) )
                                                                                ^
cc1plus.exe: all warnings being treated as errors
make[2]: *** [src/test/pegtl/CMakeFiles/pegtl-test-utf8_general.dir/build.make:63: src/test/pegtl/CMakeFiles/pegtl-test-utf8_general.dir/utf8_general.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:265: src/test/pegtl/CMakeFiles/pegtl-test-utf8_general.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

It's likely an issue with GCC 8.3.0 itself.

EDIT: I should also note that this was done with the latest release (2.7.1).

--- a/include/tao/pegtl/internal/file_mapper_win32.hpp
+++ b/include/tao/pegtl/internal/file_mapper_win32.hpp
@@ -143,7 +143,7 @@ namespace tao
 
             explicit file_mapper( const win32_file_mapper& mapper )
                : m_size( mapper.m_size ),
-                 m_data( static_cast< const char* const >(::MapViewOfFile( mapper.m_handle,
+                 m_data( static_cast< const char* >(::MapViewOfFile( mapper.m_handle,
                                                                            FILE_MAP_READ,
                                                                            0,
                                                                            0,

